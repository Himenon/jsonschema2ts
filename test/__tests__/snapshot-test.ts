import * as fs from "fs";
import * as path from "path";

describe("Generate Code Snapshot Test", () => {
  test("OnlyLocalRef.ts", () => {
    const code = fs.readFileSync(path.join(__dirname, "../output/OnlyLocalRef.ts"), { encoding: "utf-8" });
    expect(code).toMatchSnapshot();
  });
  test("RemoteRef.ts", () => {
    const code = fs.readFileSync(path.join(__dirname, "../output/RemoteRef.ts"), { encoding: "utf-8" });
    expect(code).toMatchSnapshot();
  });
});
