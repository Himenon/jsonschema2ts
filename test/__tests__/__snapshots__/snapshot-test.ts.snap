// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate Code Snapshot Test OnlyLocalRef.ts 1`] = `
"export type LocalReferenceString = string;
export type StringType = string;
export type StringHasEnumType = \\"a\\" | \\"A\\" | \\"b\\" | \\"B\\" | \\"あ\\" | \\"ア\\" | \\"い\\" | \\"イ\\" | \\"漢\\" | \\"字\\";
export type NumberType = number;
export type BooleanType = boolean;
export type ArrayType = string[];
export type OneOfType = string | number | boolean;
export type AllOfType = string & number & boolean;
export interface ObjectType {
    stringType: string;
    numberType?: number;
    booleanType: boolean;
    arrayType?: string[];
}
export type RefToDefinitionString = LocalReferenceString;
export type RefToPropertyString = StringType;
"
`;

exports[`Generate Code Snapshot Test RemoteRef.ts 1`] = `
"export interface Book {
    title?: string;
}
export type Books = Book[];
export namespace Params {
    export type B = string;
    export interface A {
        B?: B;
    }
}
export type A = Params.A;
"
`;
